### PR TITLE
[Fortran interface] also use input_schema.json for retrieving input parameters

### PR DESCRIPTION
**Pull request by mtaillefumier**
the fortran interface for requiring SIRIUS input parameters now uses Anton json dictionary. It means that the original option dictionary is obsolete and removed from the source code. It implies that  options that can take empty strings as default (with finite number of option values) now take "auto" as default. The default behavior was empty string = auto is preserved.

We might need to update the minor version of SIRIUS because it is needed for cp2k.

there is no api break so far.

### CODE REVIEW

**Review by simonpintarelli — APPROVED** [review]
lgtm

File: src/context/simulation_parameters.cpp [hunk 1]
```diff
@@ -186,46 +203,48 @@ void Simulation_parameters::processing_unit(std::string name__)
...
-                    for (size_t st = 1; st < v.size(); st++)
-                        std::cout << " " << v[st];
-                }
-                std::cout << "default value : " << dict[el.key()]["default_values"].get<std::string>() << std::endl;
-            }
-        }
-    }
-    Communicator::world().barrier();
-}
+// void Simulation_parameters::print_options() const
```
Comment (AdhocMan) [inline src/context/simulation_parameters.cpp:213]: Maybe this part could be removed instead?

**Review by AdhocMan — APPROVED** [review]
LGTM overall.

File: src/context/simulation_parameters.cpp (hunk 1 above)
Comment (mtaillefumier) [reply src/context/simulation_parameters.cpp:213]: I may have to restore the next function. I will look into it. Thanks :)

**Pull request merged**